### PR TITLE
Fix: Don't iterate forever over image list for a single tag

### PR DESCRIPTION
**Pull request by sklinkert**
...if there aren't enough images to complete all like, comment, and follow
jobs. The bot will now try to get some fresh images for tags. Furthermore,
with the new config parameter "limits.maxRetry" you are able to configure
how often the bot will try to get new images for the current tag until the
next tag will be checked. Tag limits (e.g. number of likes) in config.json
will be handled on a 'best effort' basis.

### CODE REVIEW

File: dist/config.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
         }
     },
     "limits" : {                            
+	"maxRetry" : 2,
```
Comment (tducasse) [inline dist/config.json]: Maybe we could make this optional?

File: insta.go [hunk 2]
```diff
@@ -159,11 +161,16 @@ func browse() {
 
 		goThrough(images)
 
+		if i > viper.GetInt("limits.maxRetry") {
```
Comment (tducasse) [inline insta.go]: And here, we could just check if the value is set with the **isSet** method from Viper [https://github.com/spf13/viper#getting-values-from-viper](url)

File: insta.go [hunk 3]
```diff
@@ -175,6 +182,11 @@ func goThrough(images response.TagFeedsResponse) {
 			continue
 		}
 
+		// Check if we should fetch new images for tag
+		if i >= limits["follow"] && i >= limits["like"] && i >= limits["comment"] {
```
Comment (tducasse) [inline insta.go]: This is fine by me

**Comment by sklinkert** [comment]
Thank you :)

**Pull request merged**